src/utils/management/CustomState.ts

```diff
@@ -0,0 +1,34 @@
+export class CustomState<T> {
+  private error: string;
```

Comment:
We can improve object creation feature a little bit.
Instead of new CustomState<T> = we can implement something which allows us to do CustomState<T>().

Instead of export class CustomState - remove export from class and create function.

export const State<T> = (data: T) => new CustomState<T>(data);